Cannot compile libmesh in dbg mode

Hello everyone, 

I am trying to compile libmesh in dbg mode and the compilation produces the following error 

```
  CXX      src/apps/fparser_parse_dbg-fparser_parse.o
In file included from ./include/libmesh/mesh_communication.h:26,
                 from ./include/libmesh/checkpoint_io.h:28,
                 from ../src/apps/splitter.C:24:
./include/libmesh/mesh_tools.h: In function 'void libMesh::MeshTools::libmesh_assert_valid_procids(const libMesh::MeshBase&)':
./include/libmesh/mesh_tools.h:572:3: error: 'libmesh_assert_topology_consistent_procids' was not declared in this scope; did you mean 'libmesh_assert_parallel_consistent_procids'?
  572 |   libmesh_assert_topology_consistent_procids<DofObjectSubclass>(mesh);
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   libmesh_assert_parallel_consistent_procids
./include/libmesh/mesh_tools.h:572:63: error: expected primary-expression before '>' token
  572 |   libmesh_assert_topology_consistent_procids<DofObjectSubclass>(mesh);

```


I **could** compile in opt, devel and pro. No prpblem
I am suspecting a misplaced #ifdef  but am not sure.  Is this issue known ? If not, am I missing something when compiling. 

Thank you in advance.

--- a/configure
+++ b/configure
@@ -61378,6 +61378,15 @@ $as_echo "C compiler......................... : $CC"
 $as_echo "Fortran compiler................... : $FC"
 $as_echo "Build Methods...................... : $METHODS"
 $as_echo
+if test "x$CPPFLAGS" != "x"; then :
+  $as_echo "CPPFLAGS........................... : $CPPFLAGS"
+fi
+if test "x$CXXFLAGS" != "x"; then :
+  $as_echo "CXXFLAGS........................... : $CXXFLAGS"
+fi
+if test "x$CFLAGS" != "x"; then :
+  $as_echo "CFLAGS............................. : $CFLAGS"
+fi
 for method in ${METHODS}; do
      case "${method}" in #(
   opt) :

--- a/include/base/libmesh_common.h
+++ b/include/base/libmesh_common.h
@@ -20,6 +20,13 @@
 #ifndef LIBMESH_LIBMESH_COMMON_H
 #define LIBMESH_LIBMESH_COMMON_H
 
+// These flags should never be used together.  -DDEBUG means "turn on
+// all the ridiculously expensive error checking"; -DNDEBUG means
+// "turn off all the somewhat affordable error checking"
+#if defined(DEBUG) && defined(NDEBUG)
+#  error DEBUG and NDEBUG should never be defined simultaneously
+#endif
+
 // The library configuration options
 #include "libmesh/libmesh_config.h"
 

--- a/m4/config_summary.m4
+++ b/m4/config_summary.m4
@@ -28,6 +28,12 @@ AS_ECHO(["C compiler......................... : $CC"])
 AS_ECHO(["Fortran compiler................... : $FC"])
 AS_ECHO(["Build Methods...................... : $METHODS"])
 AS_ECHO([])
+AS_IF([test "x$CPPFLAGS" != "x"],
+      [AS_ECHO(["CPPFLAGS........................... : $CPPFLAGS"])])
+AS_IF([test "x$CXXFLAGS" != "x"],
+      [AS_ECHO(["CXXFLAGS........................... : $CXXFLAGS"])])
+AS_IF([test "x$CFLAGS" != "x"],
+      [AS_ECHO(["CFLAGS............................. : $CFLAGS"])])
 for method in ${METHODS}; do
      AS_CASE("${method}",
              [opt],   [AS_ECHO(["CPPFLAGS...(opt)................... : $CPPFLAGS_OPT"])

